test/causality_tests.erl

```diff
@@ -46,54 +37,40 @@
 -define(bar(A), #{bar => A}).
 
 -define(pair(A), {pair, #{foo := A}, #{bar := A}}).
--define(singleton(A), {singleton, #{foo := A}}).
+-define(unmatched_cause(A), {unmatched_cause, #{foo := A}}).
+-define(unmatched_effect(A), {unmatched_effect, #{bar := A}}).
 
 -define(error_msg, "Effect occurs before cause").
 
--define(error_msg_cause, "Cause without effect").
+-define(error_msg_cause, "Causality violation").
 
-spng_success_test() ->
+png_success_test() ->
   ?assertMatch( []
-              , ?SPNG([])
+              , ?PNG([])
               ),
   ?assertMatch( []
-              , ?SPNG([#{quux => 1}, #{quux => 2}])
+              , ?PNG([#{quux => 1}, #{quux => 2}])
               ),
   ?assertMatch( [?pair(1), ?pair(owo)]
-              , ?SPNG([?foo(1), ?foo(owo), foo, ?bar(owo), ?bar(1), bar])
+              , ?PNG([?foo(1), ?foo(owo), foo, ?bar(owo), ?bar(1), bar])
+              ),
+  ?assertMatch( [?unmatched_cause(1), ?pair(2)]
+              , ?PNG([?foo(1), ?foo(2), foo, ?bar(2), bar])
               ),
-  ?assertMatch( [?singleton(1), ?pair(2)]
-              , ?SPNG([?foo(1), ?foo(2), foo, ?bar(2), bar])
+  ?assertMatch( [?unmatched_cause(1), ?pair(2), ?pair(owo)]
+              , ?PNG([?foo(1), ?foo(2), ?bar(2), ?foo(owo), ?bar(owo)])
               ),
-  ?assertMatch( [?singleton(1), ?pair(2), ?pair(owo)]
-              , ?SPNG([?foo(1), ?foo(2), ?bar(2), ?foo(owo), ?bar(owo)])
+  ?assertMatch( [ ?unmatched_cause(1), ?unmatched_effect(owo)
+                , ?unmatched_cause(owo), ?pair(2), ?unmatched_effect(44)
+                ]
+              , ?PNG([ ?foo(1), ?bar(owo), ?foo(owo), ?foo(2)
+                     , foo, ?bar(2), ?bar(44)
+                     ])
               ).
 
-spng_fail_test() ->
-  ?assertError( {panic, #{?snk_kind := ?error_msg}}
-              , ?SPNG([?foo(1), ?bar(owo), foo, ?foo(owo), ?bar(1), bar])
-              ).
-
-fpng_success_test() ->
-  ?assertMatch( []
-              , ?FPNG([])
-              ),
-  ?assertMatch( []
-              , ?FPNG([#{quux => 1}, #{quux => 2}])
-              ),
-  ?assertMatch( [?pair(1), ?pair(owo)]
-              , ?FPNG([?foo(1), ?foo(owo), foo, ?bar(owo), ?bar(1), bar])
-              ),
-  ?assertMatch( [?singleton(1), ?pair(2)]
-              , ?FPNG([?foo(1), ?foo(2), foo, ?bar(2), bar])
-              ),
-  ?assertMatch( [?singleton(1), ?pair(2), ?pair(owo)]
-              , ?FPNG([?foo(1), ?foo(2), ?bar(2), ?foo(owo), ?bar(owo)])
-              ),
-  ?assertMatch( [?singleton(1), ?singleton(owo), ?pair(2)]
-              , ?FPNG([ ?foo(1), ?bar(owo), ?foo(owo), ?foo(2)
-                      , foo, ?bar(2), ?bar(44)
-                      ])
+spng_unmatched_effect_test() ->
```

Comment:
`png_...`?

Reply:
I don't remember why I introduced these stupid acronyms, but they are explained here:
https://github.com/kafka4beam/snabbkaffe/pull/38/files#diff-4db3ad0ed7f05f75a8c6c822b9d0d0c285cca192440d0b8747e63798b8b0fa80R6

Reply:
Yes, it was just a nano nit: since other test case names dropped the `S` for `strict` and this one didn't :speak_no_evil:

Reply:
Thanks, I will create a follow-up